src/ibex_bluesky_core/callbacks/__init__.py

```diff
@@ -164,31 +169,41 @@ def _inner():
 
         if (add_plot_cb or show_fit_on_plot) and not ax:
             logger.debug("No axis provided, creating a new one")
-            fig, ax, exc, result = None, None, None, None
-            done_event = threading.Event()
-
-            class _Cb(QtAwareCallback):
-                def start(self, doc: RunStart) -> None:
-                    nonlocal result, exc, fig, ax
-                    try:
-                        plt.close("all")
-                        fig, ax = plt.subplots()
-                    finally:
-                        done_event.set()
-
-            cb = _Cb()
-            cb("start", {"time": 0, "uid": ""})
-            done_event.wait(10.0)
+            fig, ax = None, None
+
+            if is_matplotlib_backend_qt():
+                done_event = threading.Event()
+
+                class _Cb(QtAwareCallback):
+                    def start(self, doc: RunStart) -> None:
+                        nonlocal fig, ax
+                        try:
+                            plt.close("all")
+                            fig, ax = plt.subplots()
+                        finally:
+                            done_event.set()
+
+                cb = _Cb()
+                cb("start", {"time": 0, "uid": ""})
```

Comment:
will this end up in our raw event log? I wonder if it's worth putting some metadata saying what it's doing so we don't get confused by starts with 0 time amongst "useful" events

Reply:
no